### ISSUE TITLE
Querying by lightswitch value stopped working

### ISSUE THREAD

**Issue by gijsstegehuis** [craftcms/cms#5046]
### Description
Querying an entry by a lightswitch value stopped working for both `true` and `false` values.

### Steps to reproduce

1. Query entries by a lightswitch value. e.g. `craft.entries.section('pages').isFeatured(true)`
2. Resultset contains all possible results (both isFeatured=`true`, and isFeatured=`false`) as the generated query is `AND (NOT ("content"."field_isFeatured" IS NULL))`

### Additional info

- Craft version: 3.3.6
- PHP version:  7.3.10
- Database driver & version: PostgreSQL 11.5

**Comment by jonleverrier** [comment]
What happens when you query using:
```
craft.entries.section('pages').isFeatured('not 1')
```
```
craft.entries.section('pages').isFeatured('0')
```

**Comment by renzo-s** [comment]
Same result for 1, '1', true/false, 'not 1' etc. @jonleverrier

**Comment by jsunsawyer** [comment]
Just ran into this on my end as well. Seems that lightswitch field values always return `1`, no matter their state.

**Comment by renzo-s** [comment]
I've verified this is happening only when using Postgres.

Last tested working version is `3.3.3`, version `3.3.4` had the same issue as MySQL which was fixed in `3.3.4.1`, but from there on Postgres queries are unable to search by lightswitch at all.

Hope this can be fixed in the next release. Downgrading to `3.3.3` is a solution until that time.

**Comment by brandonkelly** [comment]
Just fixed this for the next release. We’ll get that out ASAP, but if you need it right away, change your `craftcms/cms` requirement in composer.json to:

```json
"require": {
  "craftcms/cms": "dev-develop#600e937048d83dc2ae6498a348b9f47aab064f35 as 3.3.7",
  "...": "..."
}
```

Then run `composer update`.
